### PR TITLE
Backport 5.2 changes to parser-standard

### PR DESCRIPTION
**Pull request by Julow**
AST changes, perhaps due to parser-extended not being uptodate.
This might depend on https://github.com/ocaml-ppx/ocamlformat/pull/2503

### CODE REVIEW

**Comment by gpetiot** [comment]
This only impacts `parser-standard`, which `parser-extended` doesn't depend on. This will be good to go as soon as the tests are passing.

**Comment by Julow** [comment]
The tests are not passing due AST changes, which I think are due to parser-extended not being uptodate.

**Comment by Julow** [comment]
Merged over https://github.com/ocaml-ppx/ocamlformat/pull/2519

**Comment by Julow** [comment]
This PR is changed to backport the remaining parser and lexer changes except for `Pexp_function`, which is being worked on in https://github.com/ocaml-ppx/ocamlformat/pull/2544

File: vendor/parser-standard/parser.mly [hunk 1]
```diff
@@ -1000,6 +1007,27 @@ reversed_nonempty_llist(X):
...
+| xs = reversed_nonempty_concat(X) x = X
+    { List.rev_append x xs }
+
+(* [nonempty_concat(X)] recognizes a nonempty sequence of [X]s
+   (each of which is a list), and produces an OCaml list of their concatenation
+   in direct order -- that is, the first element of the first list in the input
+   text appears first in the list.
+*)
+
+%inline nonempty_concat(X):
```
Comment (Julow) [inline vendor/parser-standard/parser.mly:1027]: This code is not used but will be in https://github.com/ocaml-ppx/ocamlformat/pull/2544

**Pull request merged**